autobahn/wamp/interfaces.py

```diff
@@ -262,7 +262,10 @@ class ISession(object):
     @abc.abstractmethod
     def onConnect(self):
         """
-        Callback fired when the transport this session will run over has been established.
+        Callback fired when the transport this session will run over has
+        been established.
+
```

Comment:
You're talking about the ability for those ISession methods to be async?

There are some instances in Crossbar that decorate at least onJoin with inlineCallbacks -- so the try...catch around self.onJoin would be insufficient to catch exceptions from such methods...